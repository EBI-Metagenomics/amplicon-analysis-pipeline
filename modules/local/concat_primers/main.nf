
process CONCAT_PRIMERS {
    // Concatenate result of both the std primer identification and auto primer identification
    // This is important in case for a run the fwd strand has a known std primer but the reverse
    // strand has an auto primer
    tag "$meta.id"
    label 'light'
    
    input:
    tuple val(meta), path(std_primers), path(auto_primers)

    output:
    tuple val(meta), path("*concat_primers.fasta") , optional: true, emit: concat_primers_out

    shell:
    '''
    if [[ -s !{auto_primers} ]]; then
        cat !{std_primers} > ./!{meta.id}_!{meta.var_region}_concat_primers.fasta
        echo '\n' >> ./!{meta.id}_!{meta.var_region}_concat_primers.fasta
        cat !{auto_primers} >>./!{meta.id}_!{meta.var_region}_concat_primers.fasta
    else
        cat !{std_primers} > ./!{meta.id}_!{meta.var_region}_concat_primers.fasta
    fi
    '''
}
