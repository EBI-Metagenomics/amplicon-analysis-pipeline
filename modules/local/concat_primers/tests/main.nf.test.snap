{
    "concat_primers_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S-V3-V4_concat_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "timestamp": "2023-11-06T13:40:59.182179511"
    }
}